packages/syft/src/syft/core/node/new/user_service.py

```diff
@@ -138,6 +147,39 @@ def update(
         if user is None:
             return SyftError(message=f"No user exists for given UID: {uid}")
 
+        if (
+            updates_role
+            and context.role == ServiceRole.DATA_OWNER
+            and user.role >= context.role
+        ):
+            return SyftError(
+                message=f"As a {context.role}, you are not allowed to edit {user.role}"
+            )
+        if (
+            updates_role
+            and context.role == ServiceRole.DATA_OWNER
+            and user_update.role >= context.role
+        ):
+            return SyftError(
+                message=f"As a {context.role}, you are not allowed to upgrade role to {user_update.role}"
+            )
+
+        edits_non_role_attrs = any(
+            [
+                getattr(user_update, attr) is not None
+                for attr in UserUpdate().__fields__.keys()
```

Comment:
An alternate way:  `user_update.to_dict(exclude_none: bool = False)` -> to get a dict of fields which are not None